### PR TITLE
Migrate bicep example: 101/aad-domainservices

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Added bicep support to this sample by integrating bicep.main from the example in https://github.com/Azure/bicep/tree/main/docs/examples/101/aad-domainservices

The corresponding PR for the modified bicep example is here: https://github.com/Azure/bicep/pull/3190

### CODE REVIEW

**Comment by StephenWeatherford** [comment]
FYI @rahulkhengare You're welcome to review if you'd like.

File: quickstarts/microsoft.network/aad-domainservices/main.bicep [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  name: domainServicesVnetName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        domainServicesVnetAddressPrefix
+      ]
+    }
+  }
+  dependsOn: [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.network/aad-domainservices/main.bicep:149]: why is this needed?

File: quickstarts/microsoft.network/aad-domainservices/metadata.json [hunk 2]
```diff
@@ -5,6 +5,6 @@
   "description": "This template deploys an Managed Azure Active Directory Domain Service with required VNet and NSG configurations.",
   "summary": "Deploy Managed AADDS with required VNet and NSG configurations",
   "githubUsername": "rahulkhengare",
-  "dateUpdated": "2021-05-14",
+  "dateUpdated": "2021-06-11",
   "validationType": "Manual"
```
Comment (bmoore-msft) [inline quickstarts/microsoft.network/aad-domainservices/metadata.json:9]: Can you remove this and see if automated testing passes?  I can't see the reason why it's needed...

If it does work, we should update the readme to remove the antiquated PS commands

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.network/aad-domainservices/main.bicep (hunk 1 above)
Comment (StephenWeatherford) [reply quickstarts/microsoft.network/aad-domainservices/main.bicep:149]: It's there in the original JSON:
```json
    {
      "apiVersion": "2020-11-01",
      "type": "Microsoft.Network/virtualNetworks",
      "name": "[parameters('domainServicesVnetName')]",
      "location": "[parameters('location')]",
      "dependsOn": ["[variables('domainServicesNSGName')]"],
      "properties": {
        "addressSpace": {
          "addressPrefixes": ["[parameters('domainServicesVnetAddressPrefix')]"]
        }
      },
      "resources": [
        {
          "apiVersion": "2020-11-01",
          "type": "subnets",
          "location": "[parameters('location')]",
          "name": "[parameters('domainServicesSubnetName')]",
          "dependsOn": ["[parameters('domainServicesVnetName')]"],
          "properties": {
            "addressPrefix": "[parameters('domainServicesSubnetAddressPrefix')]",
            "networkSecurityGroup": {
              "id": "[variables('nsgRefId')]"
            }
          }
        }
      ]
    }
```

I don't want to remove those myself cause I don't know anything about the RP structures.  if you want me to try removing it, I can, although one successful deployment doesn't prove it's not needed.  So what do you want me to do?

File: quickstarts/microsoft.network/aad-domainservices/metadata.json (hunk 2 above)
Comment (StephenWeatherford) [reply quickstarts/microsoft.network/aad-domainservices/metadata.json]: @bmoore-msft Can you look at the deployment failure?  thx

**Comment by bmoore-msft** [comment]
You need to be a tenant admin to deploy AADDS - so we need to revert the last change (to metadata.json)

File: quickstarts/microsoft.network/aad-domainservices/metadata.json (hunk 2 above)
Comment (StephenWeatherford) [reply quickstarts/microsoft.network/aad-domainservices/metadata.json:9]: reverted

File: quickstarts/microsoft.network/aad-domainservices/main.bicep (hunk 1 above)
Comment (bmoore-msft) [reply quickstarts/microsoft.network/aad-domainservices/main.bicep:149]: ah ok... the vnet doesn't have the dependency, the subnet does - but since the nsg is referenced by the subnet, bicep will create the dependson for the user... so we should remove it.

There's likely to be many more of these cases as we go through the migration.  Might be worth an informal rule to "question" any explicit dependsOn in bicep code.

Comment (StephenWeatherford) [reply quickstarts/microsoft.network/aad-domainservices/main.bicep:149]: Will remove, then.  I'm removing them when I can find clear reason.  When not, I'm leaving it in and will depend on you to tell me if it's unnecessary.  Thanks.
